cockatrice/src/settings/shortcut_treeview.h

```diff
@@ -0,0 +1,34 @@
+#ifndef SHORTCUT_TREEVIEW_H
+#define SHORTCUT_TREEVIEW_H
+
+#include <QModelIndex>
+#include <QStandardItemModel>
+#include <QTreeView>
+
+class QSortFilterProxyModel;
+class ShortcutTreeView : public QTreeView
+{
+    Q_OBJECT
+public:
+    explicit ShortcutTreeView(QWidget *parent);
+    void retranslateUi();
+
+signals:
+    void currentItemChanged(const QString &shortcut);
+
+public slots:
+    void updateSearchString(const QString &searchString);
+
+private:
+    QStandardItemModel *shortcutsModel;
+    QSortFilterProxyModel *proxyModel;
+    void populateShortcutsModel();
+
+private slots:
+    void refreshShortcuts();
+
+protected:
```

Comment:
what does setting this as protected actually do for this class that isn't subclassed again and isn't part of a library we provide (we don't provide any)?

Reply:
We’re overriding a protected method in the parent class

Reply:
ah, I see